service/matching/taskListManager.go

```diff
@@ -465,7 +473,14 @@ func (c *taskListManagerImpl) executeWithRetry(
 
 func (c *taskListManagerImpl) trySyncMatch(ctx context.Context, params addTaskParams) (bool, error) {
 	childCtx, cancel := c.newChildContext(ctx, maxSyncMatchWaitTime, time.Second)
-	task := newInternalTask(params.taskInfo, c.completeTask, params.forwardedFrom, true)
+
+	// Mocking out TaskId for syncmatch as it hasn't been allocated yet
+	fakeTaskIdWrapper := &persistenceblobs.PersistedTaskInfo{
```

Comment:
This is new behavior, previously, TaskId would be implicitly 0, now it is set to a known constant value (`-137` currently)